filter/slice_test.go

```diff
@@ -0,0 +1,174 @@
+package filter
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/AdRoll/baker"
+)
+
+func TestSlice(t *testing.T) {
+	fields := map[string]baker.FieldIndex{
+		"1st": 0,
+		"2nd": 1,
+		"3rd": 2,
+	}
+
+	fieldByName := func(s string) (baker.FieldIndex, bool) {
+		idx, ok := fields[s]
+		return idx, ok
+	}
+
+	tests := []struct {
+		name        string
+		src         string
+		dst         string
+		start       int
+		end         int
+		record      []byte
+		want        []byte
+		wantInitErr bool
+		wantErr     bool
+	}{
+		{
```

Comment:
Maybe we could add a case where both start and end are defaults, which should always copies the whole field into the destination
And another, a config error, where start > end

Reply:
ok